Gems/Atom/RPI/Code/Source/RPI.Public/Image/ImageSystem.cpp

```diff
@@ -200,6 +200,39 @@ namespace AZ
             return m_systemImages[static_cast<size_t>(simpleImage)];
         }
 
+        bool ImageSystem::RegisterAttachmentImage(const AttachmentImage& attachmentImage)
+        {
+            if (m_registeredAttachmentImages.find(attachmentImage.GetAttachmentId()) != m_registeredAttachmentImages.end())
+            {
+                AZ_Assert(false, "AttachmangeImage with name '%s' was already registered", attachmentImage.GetAttachmentId().GetCStr());
+                return false;
+            }
+
+            m_registeredAttachmentImages[attachmentImage.GetAttachmentId()] = &attachmentImage;
+
+            return true;
+        }
+
+        void ImageSystem::UnregisterAttachmentImage(const AttachmentImage& attachmentImage)
+        {
+            auto itr = m_registeredAttachmentImages.find(attachmentImage.GetAttachmentId());
+            if (itr != m_registeredAttachmentImages.end())
+            {
+                m_registeredAttachmentImages.erase(itr);
+            }
+        }
+
+        Data::Instance<AttachmentImage> ImageSystem::FindRegisteredAttachmentImage(const RHI::AttachmentId& attachmentId) const
+        {
+            auto itr = m_registeredAttachmentImages.find(attachmentId);
+
+            if (itr != m_registeredAttachmentImages.end())
+            {
+                return Data::InstanceDatabase<AttachmentImage>::Instance().Find(itr->second->GetId());
```

Comment:
Since Instance<> is just an intrusive pointer, can you just return iter->second?